File: Back/.Net/Modules/Users/BetFriends.Users.Infrastructure/UserModule.cs
using BetFriends.Shared.Application.Abstractions.Messaging;
using BetFriends.Users.Application.Abstractions;
using MediatR;

namespace BetFriends.Users.Infrastructure;

public class UserModule : IUserModule
{
    public async Task<T> ExecuteAsync<T>(IQuery<T> query)
    {
        var scope = UserCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        var result = await mediator!.Send(query);
        scope.Dispose();
        return result;
    }

    public async Task ExecuteAsync(ICommand command)
    {
        var scope = UserCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        await mediator!.Send(command);
        scope.Dispose();
    }

    public async Task ExecuteNotificationAsync(INotification notification)
    {
        var scope = UserCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        await mediator!.Publish(notification);
        scope.Dispose();
    }
}

File: Back/NodeJS/modules/users/src/infrastructure/UserModule.ts
import { Behavior } from "../../../shared/application/abstractions/Behavior";
import { IRequest } from "../../../shared/application/Request/IRequest";
import { IUserModule } from "../application/abstractions/IUserModule";

export class UserModule implements IUserModule {
    constructor(private behavior: Behavior) {}
    Execute<T>(request: IRequest<T>): Promise<T> {
        return this.behavior.Execute(request);
    }

}